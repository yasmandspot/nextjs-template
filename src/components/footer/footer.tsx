/**
 * Footer component that displays the copyright information.
 * @return {JSX.Element} The footer component.
 */
export function Footer() {
  return (
    <footer className="bg-gray-800 py-4 text-white">
      <div className="mx-auto max-w-7xl px-4 sm:px-6 lg:px-8">
        <p className="text-center text-sm">
          &copy; {new Date().getFullYear()} DSpot Next.js 15 Template. All rights reserved.
        </p>
      </div>
    </footer>
  );
}
