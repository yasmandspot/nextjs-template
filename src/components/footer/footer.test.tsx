import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { Footer } from './footer';

describe('Footer', () => {
  it('should match snapshot', () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });

  it('should display the current year in the copyright text', () => {
    render(<Footer />);
    const currentYear = new Date().getFullYear().toString();
    expect(screen.getByText(new RegExp(currentYear))).toBeInTheDocument();
  });
});
