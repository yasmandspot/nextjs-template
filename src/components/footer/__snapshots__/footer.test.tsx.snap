// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot 1`] = `
<div>
  <footer
    class="bg-gray-800 py-4 text-white"
  >
    <div
      class="mx-auto max-w-7xl px-4 sm:px-6 lg:px-8"
    >
      <p
        class="text-center text-sm"
      >
        Â© 
        2025
         DSpot Next.js 15 Template. All rights reserved.
      </p>
    </div>
  </footer>
</div>
`;
